sw/device/tests/dif/dif_uart_unittest.cc

```diff
@@ -736,5 +736,62 @@ TEST_F(TxBytesAvailableTest, FifoEmpty) {
   EXPECT_EQ(num_bytes, kDifUartFifoSizeBytes);
 }
 
+class FifoResetTest : public UartTest {};
+
+TEST_F(FifoResetTest, NullArgs) {
+  dif_uart_result_t result = dif_uart_fifo_reset(nullptr, kDifUartFifoResetRx);
+  EXPECT_EQ(result, kDifUartBadArg);
+}
+
+TEST_F(FifoResetTest, Success) {
+  EXPECT_MASK32(UART_FIFO_CTRL_REG_OFFSET, {{UART_FIFO_CTRL_RXRST, 0x1, true}});
+  dif_uart_result_t result =
+      dif_uart_fifo_reset(&dif_uart_, kDifUartFifoResetRx);
+  EXPECT_EQ(result, kDifUartOk);
+
+  EXPECT_MASK32(UART_FIFO_CTRL_REG_OFFSET, {{UART_FIFO_CTRL_TXRST, 0x1, true}});
+  result = dif_uart_fifo_reset(&dif_uart_, kDifUartFifoResetTx);
+  EXPECT_EQ(result, kDifUartOk);
+
+  EXPECT_MASK32(
+      UART_FIFO_CTRL_REG_OFFSET,
+      {
+          {UART_FIFO_CTRL_RXRST, 0x1, true}, {UART_FIFO_CTRL_TXRST, 0x1, true},
+      });
+
+  result = dif_uart_fifo_reset(&dif_uart_, kDifUartFifoResetAll);
+  EXPECT_EQ(result, kDifUartOk);
+}
+
+class LoopbackSetTest : public UartTest {};
+
+TEST_F(LoopbackSetTest, NullArgs) {
+  dif_uart_result_t result =
+      dif_uart_loopback_set(nullptr, kDifUartLoopbackSystem, kDifUartEnable);
+  EXPECT_EQ(result, kDifUartBadArg);
+}
+
+TEST_F(LoopbackSetTest, Success) {
+  EXPECT_MASK32(UART_CTRL_REG_OFFSET, {{UART_CTRL_SLPBK, 0x1, true}});
+  dif_uart_result_t result =
+      dif_uart_loopback_set(&dif_uart_, kDifUartLoopbackSystem, kDifUartEnable);
+  EXPECT_EQ(result, kDifUartOk);
```

Comment:
While this style is present in this file, googletest style is to write `EXPECT_*(expr, whatever)` directly, which allows googletest's machinery to provide better error messages. (And to be consistent with other DIF unit tests.)

You don't have to fix it in this PR but you might want to switch this over in a followup.

Reply:
The reason why I have done it this way, is that when I was converging from `bool` error codes to `dif_uart_result_t`, it was easier to switch the whole source file this way (simpler copy/paste).

I don't mind changing it later, however it will take some time, and arguably is not too important (as these are just unittests, I was proposing all along to drop standards slightly).

@lenary @gkelly what would be your take on this?

Reply:
I guess that things explicitly dictated by Google C++ Style guide must be changed, however It would be great if we could be less rigorous in unittest reviews than DIF libraries.

Reply:
It's not crucial that it gets done in this CL. I'm mostly pointing out that the style for googletest has to do with how the macros extract local source information. It's not mandatory, but it makes unit test failures easier to diagnose in my experience.

You can fix it up later. It shouldn't be too hard to do it mechanically with a regular expression or two.

>  (as these are just unittests, I was proposing all along to drop standards slightly).

I'd beware of this: unit tests are just as important as they code they test, because they're a sort of living documentation that shows what effects a function is intended to have on the system. I argue that making sure tests are readable is *more* important than the readability of an implementation.